### PR TITLE
Add zopen build --instrument option to enable code instrumentation

### PR DESCRIPTION
**Pull request by IgorTodorovskiIBM**
Leverages the zoslib implementation: https://github.com/ibmruntimes/zoslib/blob/zopen2/src/zos-instrumentation.cc (`__cyg_profile_func_enter` and `__cyg_profile_func_exit`)

### CODE REVIEW

File: bin/zopen-build [hunk 1]
```diff
@@ -794,10 +799,17 @@ setEnv()
   initDefaultEnvironment
   setDepsEnv
 
+  # xlclang pragma linkage does not work
+  if [ "${CC}" = "clang" ] && ${instrument}; then
```
[the remark this answers is not in the document]
Comment (IgorTodorovskiIBM) [reply bin/zopen-build:803]: Oh yes, good catch!

**Comment by HarithaIBM** [comment]
Where are we planning to add the documentation about visualisation of the collected metrics/info?

**Comment by IgorTodorovskiIBM** [comment]
I've added the documentation, please review when you get a chance

**Review by HarithaIBM — APPROVED** [review]

**Pull request merged**